bluesky_kafka/kafka.py

```diff
@@ -39,28 +60,52 @@ class Publisher:
     >>> RE = RunEngine({})
     >>> RE.subscribe(publisher)
     """
-    def __init__(self, address, *,
-                 serializer=pickle.dumps):
-        self.address = address
-        self.producer = Producer({'bootstrap.servers': self.address})
+
+    def __init__(
+        self, topic, bootstrap_servers, producer_config=None, serializer=pickle.dumps,
+    ):
+        self.topic = topic
+        self.producer_config = {
+            "bootstrap.servers": bootstrap_servers,
+            "enable.idempotence": True,
+            # "enable.idempotence": True is shorthand for the following configuration:
+            # "acks": "all",                              # acknowledge only after all brokers receive a message
+            # "retries": sys.maxsize,                     # retry indefinitely
+            # "max.in.flight.requests.per.connection": 5  # maintain message order when retrying
+        }
+        if producer_config is not None:
+            self.producer_config.update(producer_config)
```

Comment:
If `producer_config` includes `bootstrap.server` should this raise an error?

Reply:
To be clearer I'm thinking of the situation where the `bootstrap_severs` argument and the `producer_config['bootstrap.servers']` value disagree. Could be confusing.

Reply:
I thought about that as well.

Reply:
I think a good solution is to combine the bootstrap.servers in producer_config, if there are any. In any case the bootstrap.servers configuration should be logged.

Reply:
I see. To be clear, you mean take the union of them? That seems good to me. :+1: